entities/src/main/java/com/flipkart/varadhi/entities/LifecycleStatus.java

```diff
@@ -0,0 +1,118 @@
+package com.flipkart.varadhi.entities;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Represents the lifecycle status of an entity with its state, message, and action code.
+ * This class ensures consistency across different entity types such as topics and subscriptions.
+ */
+@Getter
+@AllArgsConstructor(onConstructor = @__(@JsonCreator))
+public class LifecycleStatus {
+
+    /**
+     * The current state of the entity.
+     */
+    private State state;
+
+    /**
+     * The message associated with the current state.
+     */
+    private String message;
+
+    /**
+     * The action code indicating the reason for the current state.
+     */
+    private ActionCode actionCode;
```

Comment:
```suggestion
    private ActorCode actorCode;
```
?

The actor of the "action" :)

Reply:
Agreed.

Reply:
Done